dash_labs/dashdown.py

```diff
@@ -0,0 +1,419 @@
+                        )
+                    except Exception as e:
+                        print(
+                            f"""
+                        The following error was generated while attempting to execute 
+                        code block number: {code_block} in file: {filename}
+                        error:  {e}                    
+                        """
+                        )
+                        pass
+
+                # side by side on large screens
+                lg = 6 if updated_props["side_by_side"] else 12
+                if updated_props["code_first"]:
+                    code_display = [
+                        dbc.Col(code_card, width=12, lg=lg),
+                        dbc.Col(app_card, width=12, lg=lg),
+                    ]
+                else:
+                    code_display = [
+                        dbc.Col(app_card, width=12, lg=lg),
+                        dbc.Col(code_card, width=12, lg=lg),
+                    ]
+                reconstructed.append(dbc.Row(code_display))
+            else:
+                reconstructed.append(
+                    dbc.Row(dbc.Col(dcc.Markdown(section, **text_markdown_props)))
+                )
+        super().__init__(reconstructed)
+
+
+def _prohibited_props_check(props, prohibited_props, code_block=None, filename=None):
+    """
+    certain props may not be updated for security reasons and/or because they are updated by
+    the MarkdownAIO component
+    """
+
+    for prop in props:
+        if isinstance(props[prop], dict):
+            for p in props[prop]:
+                if p in prohibited_props[prop]:
+                    raise Exception(
+                        f" The `{p}` prop in `{prop}` may not be user-supplied in MarkdownAIO"
+                        + f" see code block# {code_block} in module {filename}"
+                        if code_block
+                        else ""
+                    )
+
+
+def _run_code(code, scope=None, div_props=None):
+    if scope is None:
+        scope = {}
+
+    # Replacements to get this working in an `exec` environment:
+    # 1) Remove language specifier included at beginning of markdown string
+    if code.startswith("python"):
+        code = code[6:]
+    # 2) Remove the app instance in the code block otherwise app.callbacks don't work
+    code = _remove_app_instance(code)
+
+    if "app.layout" in code:
+        code = code.replace("app.layout", "layout")
+    if "layout" in code:
+        exec(code, scope)
+        return html.Div(scope["layout"], **div_props)
+
+    else:
+        # taken from https://stackoverflow.com/a/39381428
+        block = ast.parse(code, mode="exec")
+        # assumes last node is an expression
+        last = ast.Expression(block.body.pop().value)
+        exec(compile(block, "<string>", mode="exec"), scope)
+        return html.Div(eval(compile(last, "<string>", mode="eval"), scope))
```

Comment:
this is a pretty cool feature, not sure if your demos show it off. but basically allows you to have code blocks that don't even define layout. it assumes the expression is layout. something as simple as this would render a graph:

```
dcc.Graph(figure=px.scatter(...))
```

which is very similar to the jupyter world

Reply:
Yup, it's cool!  I'll add an example to the docs